extension/aten_util/test/make_aten_functor_from_et_functor_test.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/extension/aten_util/make_aten_functor_from_et_functor.h>
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/portable_type/tensor.h>
+#include <executorch/runtime/platform/runtime.h>
+#include <gtest/gtest.h>
+#include <torch/library.h>
+#include <torch/torch.h>
+
+namespace torch {
+namespace executor {
+
+using namespace ::testing;
+
+Tensor& my_op_out(const Tensor& a, Tensor& out) {
+  (void)a;
+  return out;
+}
+
+Tensor& add_1_out(const Tensor& a, Tensor& out) {
+  (void)a;
+  out.mutable_data_ptr<int32_t>()[0] += 1;
+  return out;
+}
+
+Tensor& quantized_embedding_byte_out(
+    const Tensor& weight,
+    const Tensor& weight_scales,
+    const Tensor& weight_zero_points,
+    int64_t weight_quant_min,
+    int64_t weight_quant_max,
+    const Tensor& indices,
+    Tensor& out) {
+  (void)weight;
+  (void)weight_scales;
+  (void)weight_zero_points;
+  (void)weight_quant_min;
+  (void)indices;
+  out.mutable_data_ptr<int32_t>()[0] -= static_cast<int32_t>(weight_quant_max);
+  return out;
+}
+
+class MakeATenFunctorFromETFunctorTest : public ::testing::Test {
+ public:
+  void SetUp() override {
+    torch::executor::runtime_init();
+  }
+};
+
+TEST_F(MakeATenFunctorFromETFunctorTest, Basic) {
+  auto function = WRAP(my_op_out, 1);
+  at::Tensor a = torch::tensor({1.0f});
+  at::Tensor b = torch::tensor({2.0f});
+  at::Tensor c = function(a, b);
+  EXPECT_EQ(c.const_data_ptr<float>()[0], 2.0f);
+}
+
+TORCH_LIBRARY(my_op, m) {
+  m.def("add_1.out", WRAP(add_1_out, 1));
```

Comment:
Just to make sure I understand:
- we're saying that `add_1_out` is written in the executorch style? and WRAP makes it work with the torch style?

Reply:
Yes. `WRAP` allows `add_1_out` to take `at::Tensor` instead of `torch::executor::Tensor`